OPCB/functions/tankspawner/tank_restriction.sqf

```diff
@@ -1,5 +1,22 @@
+params ["_vehicle"];
 
-	_vehicle = _this select 0;
+_vehicle addEventHandler ["GetIn",{
+	params ["_vehicle","_seat", "_player" ];
+	#include "..\..\data\vehicleDriverUnitTypes.sqf";
+	if(_seat == "driver" ) then {
+		if !(typeOf _player in _tankDriverTypes) then {
+			moveOut _player;
+		};
+	};
+}];
+_vehicle addEventHandler ["Engine",{
```

Comment:
Because they can get in the commander seat for example and switch to driver seat. That action doesn't count as "GetIn" event since they are already in

Reply:
Then you need a `"SeatSwitched"` EH for that. This is only a half solution.

Reply:
Wdym ? add another eventhandler for SeatSwitched with the same function?

Reply:
I mean keep the `GetIn` handler, get rid of the `Engine` handler and instead add a `SeatSwitched` handler to take care of people switching to unauthorised seats.